File: yggdrasil/examples/split_and_merge/src/split_and_merge_modelA.R
modelA_function <- function(in_val) {
  out_val1 <- 2 * in_val
  out_val2 <- 3 * in_val
  print(sprintf("modelA_function(%f) = (%f, %f)", in_val, out_val1, out_val2))
  return(list(out_val1,out_val2))
}

File: yggdrasil/examples/split_and_merge/src/split_and_merge_modelA.cpp
#include <iostream>

int modelA_function(float in_val, float &out_val1, float &out_val2) {
  out_val1 = 2 * in_val;
  out_val2 = 3 * in_val;
  std::cout << "modelA_function(" << in_val << ") = (" << out_val1 << ", " << out_val2 << ")" << std::endl;
  return 0;
}
